src/Microsoft.VisualStudio.Services.Agent/TraceManager.cs

```diff
@@ -71,6 +71,10 @@ private void Dispose(bool disposing)
         private Tracing CreateTraceSource(string name)
         {
             SourceSwitch sourceSwitch = Switch;
+
+            // Flush it to avoid loosing the log if the process is killed
+            Trace.AutoFlush = true;
```

Comment:
I don't think that we want this. Does this cause the listeners (e.g. file writers) to flush on every write? Ting has some thoughts on how to make the logs flush more frequently, yet keeping the buffersize the same (keeping a stopwatch, and checking if the threshold has been exceeded every time a write is called).

Curious your motivation for this change? Was it just because watching the logs, they lag behind quite a bit? Or is there a problem with a long running command and the logs potentially never getting flushed if the app is terminated?